src/Tinkoff.ISA.Scheduler/Program.cs

```diff
@@ -1,65 +1,79 @@
+                {
+                    var configuration = context.Configuration;
 
-            var logSettings = configuration.GetSection("Logging").Get<LoggingSettings>();
+                    services
+                        .AddHangfire((serviceProvider, config) =>
+                        {
+                            var mongoContext = serviceProvider
+                                .GetService<IMongoContext>();
 
-            //setup our DI
-            var serviceProvider = new ServiceCollection()
-                .AddSingleton(new LoggerFactory().AddSerilog(LogExtensions.CreateLogger(logSettings, "isa-.log")))
-                .AddLogging()
-                .AddSingleton<IJob, JiraJob>()
-                .AddSingleton<IJob, ConfluenceJob>()
-                .AddSingleton<IJob, MongoIndexingForElasticJob>()
-                .AddSingleton<IScheduler, Schedule.Scheduler>()
-                .AddSingleton<IJobsActivator, JobsActivator>()
-                .AddAppDependencies()
-                .AddDalDependencies()
-                .AddInfrastructureDependencies()
-                .Configure<SchedulerSettings>(configuration.GetSection("Scheduler"))
-                .Configure<JiraSettings>(configuration.GetSection("JiraSettings"))
-                .Configure<ConfluenceSettings>(configuration.GetSection("ConfluenceSettings"))
-                .Configure<ElasticsearchSettings>(configuration.GetSection("ElasticsearchSettings"))
-                .Configure<ConnectionStringsSettings>(configuration.GetSection("ConnectionStrings"))
-                .BuildServiceProvider();
+                            config.UseMongoStorage(
+                                mongoContext.MongoClient.Settings,
+                                "Jobs",
+                                new MongoStorageOptions
+                                {
+                                    MigrationOptions = new MongoMigrationOptions
+                                    {
+                                        Strategy = MongoMigrationStrategy.Drop,
+                                        BackupStrategy = MongoBackupStrategy.Collections
+                                    }
+                                });
+                        })
+                        .AddSingleton<IHostedService, HangfireService>()
+                        .AddAppDependencies()
+                        .AddDalDependencies()
+                        .AddInfrastructureDependencies()
+                        .Configure<ConnectionStringsSettings>(configuration.GetSection("ConnectionStrings"))
+                        .Configure<JiraSettings>(configuration.GetSection("JiraSettings"))
+                        .Configure<ConfluenceSettings>(configuration.GetSection("ConfluenceSettings"))
+                        .Configure<ElasticsearchSettings>(configuration.GetSection("ElasticsearchSettings"))
+                        .AddSingleton<IJob, JiraJob>()
```

Comment:
I think we should avoid manual registration of jobs here. There is no sense to move jobs registration in tinkoff.isa.api in this case.
Could we use [Scrutor](https://github.com/khellang/Scrutor) to automate DI registration here and in other projects?